spec/models/declaration_spec.rb

```diff
@@ -192,6 +192,77 @@ def will_change_attribute(attribute_to_change:, new_value:)
+
+        %w[started completed].each do |allowed_type|
+          context "when the declaration_type is #{allowed_type}" do
+            subject { FactoryBot.build(:declaration, training_period:, declaration_type: allowed_type) }
+
+            it { is_expected.to be_valid }
+          end
+        end
+
+        described_class.declaration_types.keys.excluding("started", "completed").each do |disallowed_type|
+          context "when the declaration_type is #{disallowed_type}" do
+            subject { FactoryBot.build(:declaration, training_period:, declaration_type: disallowed_type) }
+
+            it "is not valid" do
+              expect(subject).not_to be_valid
+              expect(subject.errors[:declaration_type]).to include("Only 'started' or 'completed' declaration types are allowed for mentor funding enabled contract periods.")
+            end
+
+            context "when the training period is for an ECT" do
+              let(:training_period) { FactoryBot.create(:training_period, :for_ect) }
+
+              it { is_expected.to be_valid }
+            end
+          end
+        end
+      end
+    end
+
+    describe "existing declarations" do
+      subject { FactoryBot.build(:declaration, :eligible, training_period:, declaration_type: :started) }
+
+      let(:training_period) { FactoryBot.create(:training_period, :for_ect) }
+
+      context "when the declaration duplicates an existing declaration" do
+        before { FactoryBot.create(:declaration, :payable, training_period:, declaration_type: :started) }
+
+        it "is not valid" do
+          expect(subject).not_to be_valid
+          expect(subject.errors[:base]).to include("A matching declaration already exists.")
+        end
+      end
+
+      context "when the declaration does not duplicate an existing declaration" do
+        before { FactoryBot.create(:declaration, :voided, training_period:, declaration_type: :completed) }
+
+        it { is_expected.to be_valid }
+      end
+    end
+  end
+
+  describe "scopes" do
+    let!(:no_payment_declaration) { FactoryBot.create(:declaration, :no_payment) }
+    let!(:eligible_declaration) { FactoryBot.create(:declaration, :eligible) }
+    let!(:payable_declaration) { FactoryBot.create(:declaration, :payable) }
+    let!(:paid_declaration) { FactoryBot.create(:declaration, :paid) }
+    let!(:voided_declaration) { FactoryBot.create(:declaration, :voided) }
+    let!(:ineligible_declaration) { FactoryBot.create(:declaration, :ineligible) }
+    let!(:awaiting_clawback_declaration) { FactoryBot.create(:declaration, :awaiting_clawback) }
+    let!(:clawed_back_declaration) { FactoryBot.create(:declaration, :clawed_back) }
+
+    describe ".billable_or_changeable" do
+      subject { described_class.billable_or_changeable }
+
+      it { is_expected.to include(no_payment_declaration, eligible_declaration, payable_declaration, paid_declaration) }
```

Comment:
Could we use `contain_exactly` here? I think `include` leaves a little ambiguity (i.e are there other records we haven't explicitly expected?).

What do you think?

```suggestion
      it { is_expected.to contain_exactly(no_payment_declaration, eligible_declaration, payable_declaration, paid_declaration) }
```

Reply:
Yep good spot 👍